guidelines/component-usage.md

```diff
@@ -1 +1,106 @@
+## Table of content
 
+1. [Syntax](#syntax)
+2. [Formatting](#formatting)
+3. [Component Structure](#component-structure)
+4. [Eslint](#eslint)
+
+
+## Syntax
+
+1. #### Naming Style **[⬆](#table-of-content)**
+
+   * Always use `.jsx` | `.tsx` extensions.
```

Comment:
Keep in mind MusicBrainz uses Flow, so any shared components will have to be plain JS.  I think the document should mention this.  Otherwise at a glance this seems to encourage writing shared components in TypeScript (same for several bullet points below).

This doesn't mean we can't include TS or Flow types in the repo, BTW.  With Flow it's trivial to define types in an [adjacent .js.flow file](https://flow.org/en/docs/declarations/), and I assume something similar is possible with TS.

Re: the actual extensions, in MB we stick with the .js extension everywhere, but I have no problem with .jsx in this repo.

Reply:
Fixed